### PR TITLE
fix(android): don't crash on app exit when detach() runs twice

### LINKED ISSUE #199: Android: app crashes on every exit — detach() runs twice, second call throws in ResourceFileCache
Every exit from an app that uses this plugin crashes on Android: `detach()` runs twice during shutdown and the second run throws.

`FlutterReadiumPlugin` calls `ReadiumReader.detach()` from two callbacks:

- `onDetachedFromActivity()` (FlutterReadiumPlugin.kt:118)
- `onDetachedFromEngine()` (FlutterReadiumPlugin.kt:70)

Both fire when the activity is destroyed. The first call clears `appRef`, the second one still enters `closePublication()`, which ends in `ResourceFileCache.purgeAll()` → `directory` → `ReadiumReader.application` → `IllegalStateException`. It happens inside `onDestroy`, so it is fatal.

Version 0.3.3, also present on `main` (checked today). Android 16 emulator, Flutter 3.35, but nothing here looks device-specific.

```
E/AndroidRuntime: FATAL EXCEPTION: main
E/AndroidRuntime: Process: pl.adrio, PID: 14134
E/AndroidRuntime: java.lang.RuntimeException: Unable to destroy activity {pl.adrio/pl.adrio.MainActivity}
E/AndroidRuntime:  at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:6647)
E/AndroidRuntime: Caused by: java.lang.IllegalStateException: Application not initialized. Call ReadiumReader.attach(...) first.
E/AndroidRuntime:  at dk.nota.flutterreadium.ReadiumReader.getApplication(ReadiumReader.kt:498)
E/AndroidRuntime:  at dk.nota.flutterreadium.ResourceFileCache.getDirectory(ResourceFileCache.kt:17)
E/AndroidRuntime:  at dk.nota.flutterreadium.ResourceFileCache.purgeAll(ResourceFileCache.kt:38)
E/AndroidRuntime:  at dk.nota.flutterreadium.ReadiumReader.closePublication(ReadiumReader.kt:783)
E/AndroidRuntime:  at dk.nota.flutterreadium.ReadiumReader.detach(ReadiumReader.kt:459)
E/AndroidRuntime:  at dk.nota.flutterreadium.FlutterReadiumPlugin.onDetachedFromEngine(FlutterReadiumPlugin.kt:72)
E/AndroidRuntime:  at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.remove(FlutterEngineConnectionRegistry.java:274)
E/AndroidRuntime:  at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onDetach(FlutterActivityAndFragmentDelegate.java:811)
```

**Steps to reproduce:** open any app that registers the plugin (opening a publication is not required — the log above comes from a session where the reader had already been closed), then close it with the back gesture. The `PluginLog` lines `onDetachedFromActivity` / `onDetachedFromEngine` appear right before the crash.

**Suggested fix:** make `detach()` a no-op when nothing is attached any more. PR: (linked below)

An alternative would be to guard `ResourceFileCache` instead (return early when the application reference is gone), which also covers a `closePublication()` arriving from Dart after detach. Happy to switch the PR to that shape if you prefer it.

Thanks for the plugin — the reader itself works well for us.

### PR DESCRIPTION
**Pull request by bparuzel**
Fixes #199.

`FlutterReadiumPlugin` calls `ReadiumReader.detach()` from both `onDetachedFromActivity()` and `onDetachedFromEngine()`. Both fire when the activity is destroyed, so the second call runs after `appRef` has been cleared, enters `closePublication()` and throws `IllegalStateException` from `ResourceFileCache.purgeAll()` — inside `onDestroy`, so it kills the app on every exit.

This returns early when neither the application nor the activity reference is left, which is exactly the state the second call sees. Nothing is released twice and the first call is unchanged.

Happy to move the guard into `ResourceFileCache` instead if you would rather have it cover a `closePublication()` arriving from Dart after detach — see the note in the issue.

### CODE REVIEW

**Review by m-abs — APPROVED** [review]
Nice catch, thank you.

**Comment by m-abs** [comment]
@ddfreiling Do you have any objections to merging this?

**Review by ddfreiling — APPROVED** [review]
Looks fine to me.
